### PR TITLE
feat(cli): check database connection in debug command

### PR DESCRIPTION
**Pull request by erie0210**
Hi,

Even though @liang799 did his nice [PR work ](https://github.com/mikro-orm/mikro-orm/pull/3874), I tried mine, too 😹. As I knew we already have PR, I just want to get a review if it is okay. 

While working with this PR, I wanted to ask a question. Is it okay to fix existing test? As I added information to dump, it crashed with existing test. So I fixed them test but doesn't feel right.. 👀.. 

Thank you!

### CODE REVIEW

**Review by B4nan — COMMENTED** [review]
this one is much more in line with what i was expecting, still i believe it wont work for the same reason as the other PR - you need to connect explicitly, otherwise, it will never connect

> While working with this PR, I wanted to ask a question. Is it okay to fix existing test? 

Yeah that's totally fine, you changed behavior, so you need to adjust some assertions.

**Comment by erie0210** [comment]
@B4nan  May I ask you reviewing the fix? 🙌
Additionally, Can you recommend  other issues to try?

**Review by rubiin — CHANGES_REQUESTED** [review]
might also need some changes on documentation for the website

File: packages/cli/src/commands/DebugCommand.ts [hunk 1]
```diff
@@ -27,8 +28,15 @@ export class DebugCommand implements CommandModule {
     try {
       const config = await CLIHelper.getConfiguration();
       CLIHelper.dump(` - configuration ${colors.green('found')}`);
-      const tsNode = config.get('tsNode');
 
+      const isConnected = await CLIHelper.isDBConnected();
+      if (isConnected) {
+        CLIHelper.dump(`   - ${colors.green('database connected')}`);
+      } else {
+        CLIHelper.dump(`   - ${colors.yellow('database not found')}`);
```
Comment (B4nan) [inline packages/cli/src/commands/DebugCommand.ts:36]: while i do understand the additional indent, i'd rather keep it on the same level as other checks, as they are also about the configuration...

also, lets improve the messages a bit, the cause might be different than db not existing. latest versions also ensure the database existing during the init phase, so unless you dont have some other problem in the config, this should never happen

```suggestion
        CLIHelper.dump(` - ${colors.green('database connection succesful')}`);
      } else {
        CLIHelper.dump(` - ${colors.yellow('database connection failed')}`);
```

File: tests/features/cli/CreateDatabaseCommand.test.ts [hunk 2]
```diff
@@ -7,7 +7,8 @@ import { CreateDatabaseCommand } from '../../../packages/cli/src/commands/Create
 import { initORMSqlite } from '../../bootstrap';
 
 const closeSpy = jest.spyOn(MikroORM.prototype, 'close');
-const showHelpMock = jest.spyOn(CLIHelper, 'showHelp');
+const showHelpMock =
+  jest.spyOn(CLIHelper, 'showHelp');
```
Comment (B4nan) [inline tests/features/cli/CreateDatabaseCommand.test.ts:11]: next time please dont make unrelated changes
```suggestion
const showHelpMock = jest.spyOn(CLIHelper, 'showHelp');
```

File: tests/features/cli/DebugCommand.test.ts [hunk 3]
```diff
@@ -33,6 +38,7 @@ describe('DebugCommand', () => {
       [' - searched config paths:'],
       [`   - ${Utils.normalizePath(process.cwd() + '/path/orm-config.ts') } (found)`],
       [' - configuration found'],
+      ['   - database connected'],
```
Comment (B4nan) [inline tests/features/cli/DebugCommand.test.ts:41]:
```suggestion
      [' - database connection succesful'],
```

File: tests/features/cli/DebugCommand.test.ts [hunk 4]
```diff
@@ -47,6 +53,7 @@ describe('DebugCommand', () => {
...
```
Comment (B4nan) [inline tests/features/cli/DebugCommand.test.ts:56]:
```suggestion
      [' - database connection succesful'],
```

File: tests/features/cli/DebugCommand.test.ts [hunk 5]
```diff
@@ -66,6 +73,7 @@ describe('DebugCommand', () => {
...
```
Comment (B4nan) [inline tests/features/cli/DebugCommand.test.ts:76]:
```suggestion
      [' - database connection succesful'],
```

File: tests/features/cli/DebugCommand.test.ts [hunk 6]
```diff
@@ -92,10 +100,27 @@ describe('DebugCommand', () => {
       [' - searched config paths:'],
       [`   - ${Utils.normalizePath(process.cwd() + '/path/orm-config.ts') } (not found)`],
       [' - configuration found'],
+      ['   - database connected'],
```
Comment (B4nan) [inline tests/features/cli/DebugCommand.test.ts:103]:
```suggestion
      [' - database connection succesful'],
```

File: tests/features/cli/DebugCommand.test.ts [hunk 7]
```diff
@@ -92,10 +100,27 @@ describe('DebugCommand', () => {
...
+    const connectionMock = jest.spyOn(CLIHelper, 'isDBConnected');
+    connectionMock.mockImplementation(async () => false);
+    await expect(cmd.handler()).resolves.toBeUndefined();
+    expect(dumpDependencies).toBeCalledTimes(6);
+    expect(dump.mock.calls).toEqual([
+      ['Current MikroORM CLI configuration'],
+      [' - searched config paths:'],
+      [`   - ${Utils.normalizePath(process.cwd() + '/path/orm-config.ts') } (not found)`],
+      [' - configuration found'],
+      ['   - database not found'],
```
Comment (B4nan) [inline tests/features/cli/DebugCommand.test.ts:122]:
```suggestion
      [' - database connection failed'],
```

**Comment by B4nan** [comment]
Thanks!

**Pull request merged**